### PR TITLE
Custom CUDA operation (optional) + bug fix of DPointNet training

### PR DESCRIPTION
**Pull request by shixnya**
## Summary

This PR adds an optional fused CUDA implementation for DPointNet synaptic-current computation and fixes several mixed-precision training issues.

This change speeds up the comuputation ~50% and saves ~20% memory. It requires compiling C++ code.

### Fused CUDA operation

- Adds CSR-based TensorFlow CUDA operations for recurrent and input synaptic currents.
- Implements float16 and float32 forward and gradient kernels.
- Supports uint32 and int64 connectivity metadata.
- Preserves spike counts for input currents and computes gradients only where required.
- Validates GPU compute capability and requires exactly one visible GPU.
- Releases CUDA connectivity resources during model rebuild and cleanup.

The feature is opt-in through `use_fused_cuda`:

- `false`: use the TensorFlow implementation.
- `true`: require the fused CUDA operation.
- `"auto"`: use CUDA when available, otherwise fall back to TensorFlow.

### Training fixes

- Corrects Keras 3 loss-scaling detection. A regular optimizer's no-op `scale_loss()` method is no longer mistaken for active mixed-precision loss scaling.
- Synchronizes recurrent and trainable input compute-weight shadows after optimizer updates.
- Reorders fused CSR weight shadows whenever master weights change.
- Prevents uint32 overflow while calculating lexicographic edge-sort keys.

### Build and packaging

The operation can be built from the active environment with:

```bash
python -m bmtk.simulator.dpointnet.custom_ops.build

### CODE REVIEW

File: bmtk/simulator/dpointnet/cell_models/glif3_cell.py [hunk 1]
```diff
@@ -410,6 +417,12 @@ def straight_through_dampen(x, dampening):
     return x * (tf_one - dampening) + tf.stop_gradient(x * dampening)
 
 
+def _validate_fused_cuda_option(value):
+    if value is True or value is False or value == 'auto' and isinstance(value, str):
```
Comment (kaeldai) [inline bmtk/simulator/dpointnet/cell_models/glif3_cell.py]: Is there a reason it has to be a python "str" and not also be a byte, unicode, numpy or other type of string (many of which will fail). For instance sometimes if the value is read in from a npz or hdf5 file you can have _fused == 'auto' and still fail.
